### PR TITLE
[wip] logging socket data

### PR DESCRIPTION
**Pull request by mikz**
it is quite useful when developing, debugging, ...
will try to add more places and make the logger configurable

wdyt?

### CODE REVIEW

**Comment by dblock** [comment]
I like it.

I wouldn't log the method names by interpolating strings, different people want different styles of what is being logged, so I would just do `logger.debug buffer` and not try to impose a format on anyone.

**Comment by mikz** [comment]
@dblock then I'd go with `logger.debug('[slack#write]') { buffer }`. You can ignore the prefix by setting log format. 
Some of the buffers are binary, and without knowing from where they come, it is madness.

**Comment by dblock** [comment]
:+1: 

I think logger then should become part of `Slack::RealTime::Socket`, and so on and so forth and be documented.

**Comment by mikz** [comment]
@dblock pushed more stuff. It is WIP. 
Two parts I'm not happy with:

* eventmachine driver instrumentation
* the code to print events

The later could be fixed, but the eventmachine don't think so.

File: lib/slack/real_time/concurrency/eventmachine.rb [hunk 1]
```diff
@@ -30,7 +30,23 @@ def ensure_reactor_running
           end
 
           def connect
-            @driver = ::Faye::WebSocket::Client.new(url, nil, options)
```
Comment (dblock) [inline lib/slack/real_time/concurrency/eventmachine.rb]: Did you try inheriting from `Faye::WebSocket::Client` and overriding `parse` and `write` methods? And instantiate the child class here?

**Comment by dblock** [comment]
I see what you mean wrt Eventmachine, see my suggestion above.

**Comment by dblock** [comment]
I want to release 0.6.0 soon, maybe you can find a bit of time to finish this? Can I help?

**Comment by mikz** [comment]
@dblock will spend some time on it on wednesday and thursday.

**Comment by dblock** [comment]
Yeah, I'll do a release before that.

**Comment by dblock** [comment]
I am still looking forward to this @mikz!

**Comment by mikz** [comment]
@dblock whenever I'll do another round of bot development, I'll definitely do this.

**Comment by dblock** [comment]
Closing in favor of https://github.com/dblock/slack-ruby-client/pull/67. Fixed tests, made some improvements, still a few TODOs remaining.

**Pull request closed without merging**